lib/core/error.js

```diff
@@ -83,9 +88,28 @@ class MongoError extends Error {
  * @extends MongoError
  */
 class MongoNetworkError extends MongoError {
-  constructor(message) {
+  constructor(message, options) {
     super(message);
     this.name = 'MongoNetworkError';
+
+    if (options && !!options.beforeHandshake) {
```

Comment:
As I understand it, `!!` is used to coerce a value into boolean based on its truthiness, i.e. `!!value` is equivalent to `value ? true : false`. Is there a purpose to doing this as part of a logical expression inside an `if` condition? I'd think this `if` should behave the same with and without the `!!`.

Reply:
I don't think it matters all too much how this is written, but perhaps a `options && options.beforeHandshake === true` is the best approach, I'll make the change